src/api/che-api-client.ts

```diff
@@ -0,0 +1,343 @@
+      }
+    } catch {
+      throw new Error(`E_CHE_API_URL_NO_RESPONSE - Failed to connect to "${this.cheApiUrl}". Is it the right url?`)
+    }
+    throw new Error(`E_CHE_API_WRONG_URL - Provided url "${this.cheApiUrl}" is not Che API url`)
+  }
+
+  async isCheServerReady(responseTimeoutMs = this.defaultCheResponseTimeoutMs): Promise<boolean> {
+    const id = this.axios.interceptors.response.use(response => response, async (error: any) => {
+      if (error.config && error.response && (error.response.status === 404 || error.response.status === 503)) {
+        await sleep(500)
+        return this.axios.request(error.config)
+      }
+      return Promise.reject(error)
+    })
+
+    try {
+      await this.axios.get(`${this.cheApiUrl}/system/state`, { timeout: responseTimeoutMs })
+      return true
+    } catch {
+      return false
+    } finally {
+      this.axios.interceptors.response.eject(id)
+    }
+  }
+
+  async getCheServerStatus(responseTimeoutMs = this.defaultCheResponseTimeoutMs): Promise<string> {
+    const endpoint = `${this.cheApiUrl}/system/state`
+    let response = null
+    try {
+      response = await this.axios.get(endpoint, { timeout: responseTimeoutMs })
+    } catch (error) {
+      throw this.getCheApiError(error, endpoint)
+    }
+    if (!response || response.status !== 200 || !response.data || !response.data.status) {
+      throw new Error('E_BAD_RESP_CHE_API')
+    }
+    return response.data.status
+  }
+
+  async startCheServerShutdown(accessToken = '', responseTimeoutMs = this.defaultCheResponseTimeoutMs): Promise<void> {
+    const endpoint = `${this.cheApiUrl}/system/stop?shutdown=true`
+    const headers = accessToken ? { Authorization: accessToken } : null
+    let response = null
+    try {
+      response = await this.axios.post(endpoint, null, { headers, timeout: responseTimeoutMs })
+    } catch (error) {
+      if (error.response && error.response.status === 409) {
+        return
+      } else {
+        throw this.getCheApiError(error, endpoint)
+      }
+    }
+    if (!response || response.status !== 204) {
+      throw new Error('E_BAD_RESP_CHE_API')
+    }
+  }
+
+  async waitUntilCheServerReadyToShutdown(intervalMs = 500, timeoutMs = 60000): Promise<void> {
+    const iterations = timeoutMs / intervalMs
+    for (let index = 0; index < iterations; index++) {
+      let status = await this.getCheServerStatus()
+      if (status === 'READY_TO_SHUTDOWN') {
+        return
+      }
+      await cli.wait(intervalMs)
+    }
+    throw new Error('ERR_TIMEOUT')
+  }
+
+  /**
+   * Returns list of all workspaces of the user.
+   */
+  async getAllWorkspaces(accessToken?: string): Promise<chetypes.workspace.Workspace[]> {
+    const all: any[] = []
+    const maxItems = 30
+    let skipCount = 0
+
+    do {
+      const workspaces = await this.getWorkspaces(skipCount, maxItems, accessToken)
+      all.push(...workspaces)
+      skipCount += workspaces.length
+    } while (all.length === maxItems)
```

Comment:
I think here should be `} while (workspaces.length === maxItems)` to retrieve really all workspaces... Current logic could retrieve no more than 60 workspaces....